fe/fe-core/src/main/java/org/apache/doris/common/Config.java

```diff
@@ -1301,4 +1301,13 @@
      */
     @ConfField
     public static boolean enable_alpha_rowset = false;
+
+    /**
+     * This config is used to solve fe heartbeat response read_timeout problem,
```

Comment:
Thrift protocol doesn't have the timeout issue?

Reply:
for non master fe, the thread for processing request in fe thrift server is usually idle, and thrift protocol is more lightweight when we keep a long connection opened. But I still confused that why heartbeat by http is easy to readtimout, we just made period http request to fe server, such metric collect,  count running transaction in every db.

Reply:
When HTTP heartbeat read timeout, the FE server has a GC issue?
I think HTTP and thrift protocol are both lightweight, and the heartbeat logic is very simple. So, If the http heartbeat has a timeout issue, there is no reason thrift heartbeat won't have the timeout issue.

Reply:
There is no GC issue in fe server. All seems normal, but when non master fe run after sometimes, the read timeout problem occur, and when we restart non master fe, the read timeout issue disappear, and we are not sure that when to meet the read timeout problem again.

Reply:
2020-12-07 14:30:59,424 INFO 115 [RestBaseAction.handleRequest():52] receive http request. url=/api/bootstrap?
2020-12-07 14:31:04,431 INFO 128 [RestBaseAction.handleRequest():52] receive http request. url=/api/bootstrap?
2020-12-07 14:31:09,436 INFO 141 [RestBaseAction.handleRequest():52] receive http request. url=/api/bootstrap
2020-12-07 14:31:14,442 INFO 148 [RestBaseAction.handleRequest():52] receive http request. url=/api/bootstrap
2020-12-07 14:31:19,449 INFO 77 [RestBaseAction.handleRequest():52] receive http request. url=/api/bootstrap?
2020-12-07 14:31:24,455 INFO 87 [RestBaseAction.handleRequest():52] receive http request. url=/api/bootstrap
2020-12-07 14:31:29,461 INFO 94 [RestBaseAction.handleRequest():52] receive http request. url=/api/bootstrap?
2020-12-07 14:31:34,466 INFO 104 [RestBaseAction.handleRequest():52] receive http request. url=/api/bootstrap?
2020-12-07 14:31:39,472 INFO 119 [RestBaseAction.handleRequest():52] receive http request. url=/api/bootstrap?
2020-12-07 14:31:44,477 INFO 129 [RestBaseAction.handleRequest():52] receive http request. url=/api/bootstrap?
2020-12-07 14:31:53,897 INFO 137 [RestBaseAction.handleRequest():52] receive http request. url=/api/bootstrap?
2020-12-07 14:31:56,540 INFO 150 [RestBaseAction.handleRequest():52] receive http request. url=/api/bootstrap?
2020-12-07 14:32:01,545 INFO 79 [RestBaseAction.handleRequest():52] receive http request. url=/api/bootstrap?
2020-12-07 14:32:06,550 INFO 89 [RestBaseAction.handleRequest():52] receive http request. url=/api/bootstrap?
2020-12-07 14:32:11,556 INFO 96 [RestBaseAction.handleRequest():52] receive http request. url=/api/bootstrap

The heartbeat logic is so simple and no blocked logic to process,  but we find receive request has delayed somtimes  according to the log info(every 5 seconds should receive the heartbeat request)

Reply:
I see. Thanks you.